### ISSUE TITLE
Idea: add url base for subfolder proyect: http:/127.0.0.1/sub/proyect/router.php

### ISSUE THREAD

**Issue by soportesistemas4** [phprouter/router_xss_csrf#17]
diff --git a/router.php b/router.php
index 1ff3ccd..73221f1 100644
--- a/router.php
+++ b/router.php
@@ -1,7 +1,10 @@
 <?php
-
 session_start();
 
+if(!function_exists('base_url')){
+	function base_url(){ return '';}
+}
+
 function get($route, $path_to_include){
   if( $_SERVER['REQUEST_METHOD'] == 'GET' ){ route($route, $path_to_include); }  
 }
@@ -19,7 +22,7 @@ function delete($route, $path_to_include){
 }
 function any($route, $path_to_include){ route($route, $path_to_include); }
 function route($route, $path_to_include){
-  $ROOT = $_SERVER['DOCUMENT_ROOT'];
+  $ROOT = __DIR__;
   if($route == "/404"){
     include_once("$ROOT/$path_to_include");
     exit();
@@ -27,11 +30,12 @@ function route($route, $path_to_include){
   $request_url = filter_var($_SERVER['REQUEST_URI'], FILTER_SANITIZE_URL);
   $request_url = rtrim($request_url, '/');
   $request_url = strtok($request_url, '?');
+  $request_url = str_replace(base_url(), '', $request_url);//delete subfolders
   $route_parts = explode('/', $route);
   $request_url_parts = explode('/', $request_url);
   array_shift($route_parts);
   array_shift($request_url_parts);
diff --git a/routes.php b/routes.php
index 8fa2377..6b7d2bb 100644
--- a/routes.php
+++ b/routes.php
@@ -1,6 +1,7 @@
 <?php
+function base_url(){ return '/dev-test/main';}
 
-require_once("{$_SERVER['DOCUMENT_ROOT']}/router.php");
+require_once(__DIR__ . "/router.php");
 
 // ##################################################
 // ##################################################

**Comment by phprouter** [comment]
the library is expected to work under the main htdocs folder and for 1 project at a time. Therefore, no need to support multiple projects in sub-folders.

**Comment by CarterSnich** [comment]
Putting the library in a sub folder, like `http://localhost/project/router.php`, also won't gonna work? It's not working on XAMPP. I'm still a learner and I'm still far from deploying a website of my own server, so I have multiple projects inside my `htdocs` folder.

I'm trying to understand your library, and this is what I get. I have created an index and 404 page, but only 404 works and it seems like it's looping through all routes added.

This is what I got.
![image](https://user-images.githubusercontent.com/52433531/165668208-575949bb-bd24-4c32-8ca6-dd0bfdbf246e.png)

I only created pages for `/` and `404`, also removed the comments so it could fit on the screenshot.
![image](https://user-images.githubusercontent.com/52433531/165668292-227cd729-260a-4d6b-9b80-5bcced5135ca.png)

Folder structure.
![image](https://user-images.githubusercontent.com/52433531/165668391-f514e3ce-35f7-4ffb-a803-41826f328ab7.png)
